### PR TITLE
fix(console): update api details

### PR DESCRIPTION
**Pull request by illichso**
## Issue

https://gravitee.atlassian.net/browse/APIM-6851

## Description

A newly added information on API Details section is displayed only after refreshing the page

## Additional context

https://github.com/user-attachments/assets/69bf0701-3164-40a4-8ad5-350219bb6c53

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-yfivkqtwim.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.component.spec.ts [hunk 1]
```diff
@@ -82,20 +117,28 @@ describe('PortalApiListComponent', () => {
   }
   const init = async (apiPortalHeaders: ApiPortalHeader[]) => {
     await TestBed.configureTestingModule({
-      imports: [GioTestingModule, NoopAnimationsModule, PortalBannerComponent],
+      imports: [MatIconTestingModule, NoopAnimationsModule, GioTestingModule, PortalBannerComponent],
```
Comment (jourdiw) [inline gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.component.spec.ts]: The `MatIconTestingModule` is already in the `GioTestingModule`, so you can revert this change 👍

File: gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.component.spec.ts [hunk 2]
```diff
@@ -125,17 +168,39 @@ describe('PortalApiListComponent', () => {
...
+    httpTestingController
+      .expectOne({
+        url: `${CONSTANTS_TESTING.env.baseURL}/configuration/apiheaders/id1`,
+        method: 'DELETE',
+      })
+      .flush(null);
+
+    configureApiHeaders(fakeApiPortalHeadersAfterDelete);
+    expect(fakeSnackBarService.success).toHaveBeenCalledWith("API Information 'team' deleted successfully");
+    configureApiHeaders(fakeApiPortalHeadersAfterDelete);
```
Comment (jourdiw) [inline gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.component.spec.ts]: Hmm why is the GET being called twice?

File: gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.component.spec.ts [hunk 3]
```diff
@@ -125,17 +168,39 @@ describe('PortalApiListComponent', () => {
...
+    await dia.confirm();
+
+    httpTestingController
+      .expectOne({
+        url: `${CONSTANTS_TESTING.env.baseURL}/configuration/apiheaders/id1`,
+        method: 'DELETE',
+      })
+      .flush(null);
+
+    configureApiHeaders(fakeApiPortalHeadersAfterDelete);
```
Comment (jourdiw) [inline gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.component.spec.ts:173]: You are not checking the data returned by this call (which is the right thing to do), so you can revert changing the variable `fakeApiPortalHeaders`. As a rule for bug fixes, it's best practice to include as little changes as possible to avoid conflicts on other branches. 

Side note:
I do not think you should be checking the flushed data because you would ultimately be testing that the flush returned mock data. So you can keep the test like this and just revert the global variable changes 👍

File: gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.harness.ts [hunk 4]
```diff
@@ -95,4 +102,27 @@ export class PortalApiListHarness extends ComponentHarness {
...
+      .then((rows) => rows[index])
+      .then((row) =>
+        Promise.all([row.getCells({ columnName: 'name' }), row.getCells({ columnName: 'value' }), row.getCells({ columnName: 'actions' })]),
+      )
+      .then(([names, value, actions]) =>
+        Promise.all([
+          names[0].getText(),
+          value[0].getText(),
+          actions[0].getHarness(MatButtonHarness.with({ selector: '[data-testid="edit-button"]' })).catch((_) => undefined),
+          actions[0].getHarness(MatButtonHarness.with({ selector: '[data-testid="delete-button"]' })).catch((_) => undefined),
```
Comment (jourdiw) [inline gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.harness.ts]:
```suggestion
          actions[0].getHarnessOrNull(MatButtonHarness.with({ selector: '[data-testid="edit-button"]' })),
          actions[0].getHarnessOrNull(MatButtonHarness.with({ selector: '[data-testid="delete-button"]' })),
```

File: gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.component.spec.ts (hunk 1 above)
Comment (illichso) [reply gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.component.spec.ts]: Done

File: gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.component.spec.ts (hunk 2 above)
Comment (illichso) [reply gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.component.spec.ts]: UPdated

File: gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.harness.ts (hunk 4 above)
Comment (illichso) [reply gravitee-apim-console-webui/src/portal/customization/api/api-list/portal-api-list.harness.ts]: Done

**Review by phiz71 — APPROVED** [review]
LGTM.
To simplify the code, maybe we could have used something like this:

`fakeApiPortalHeadersAfterDelete = initialFakeApiPortalHeaders.shift()`

To avoid rewriting the same data. But it's not a blocking thing. Up to you to change before merging if you want.
